### PR TITLE
Feature: Signup Page

### PR DESCRIPTION
**Pull request by analou-mae**
Added Files: 
src/pages/auth/Signup.jsx
src/constants/auth/index.js

Updated Files:
src/pages/auth/Signin.jsx
src/utils/index.js
src/routes/index.js
src/App.jsx
src/constants/routes/index.js

### CODE REVIEW

File: src/constants/auth/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const CONFIRM_PASSOWRD_ERROR = 'Passwords do not match.';
```
Comment (AccentureJace) [inline src/constants/auth/index.js]: Follow this naming convention

export const VALIDATION_CONFIRM_PASSWORD
export const VALIDATION_FIRST_NAME

export const SUCCESS_SIGNUP

export const VALIDATION_USERNAME

File: src/pages/auth/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+const Signup = () => {
+    const navigate = useNavigate();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [form] = Form.useForm();
+
+    const handleSubmitForm = async (values) => {
+        setIsSubmitting(true);
+
+        const validation_errors = validateObject(values);
```
Comment (AccentureJace) [inline src/pages/auth/Signup.jsx]: Does the global validations applicable to this one or we need separate validations?

File: src/utils/index.js [hunk 3]
```diff
@@ -58,3 +60,13 @@ export const handleFormatAmountToPHP = (amount) => {
 		maximumFractionDigits: 2,
 	}).format(amount);
 };
+
+export const handleConfirmPassword = (form, _) => {
+    return (_, value) => {
+        const password = form.getFieldValue('password');
+        if (value && value !== password) {
+            return Promise.reject(new Error(CONFIRM_PASSOWRD_ERROR));
```
Comment (AccentureJace) [inline src/utils/index.js]: follow the variable name in the constants for consistency

Comment (AccentureJace) [inline src/utils/index.js]: use async await instead of Promise

**Review by AccentureJace — CHANGES_REQUESTED** [review]
Please see request changes

File: src/constants/auth/index.js [hunk 4]
```diff
@@ -1 +1,13 @@
-
+export const VALIDATION_CONFIRM_PASSWORD = 'Passwords do not match.';
+export const VALIDATION_FIRST_NAME = 'Please enter valid first name';
+export const VALIDATION_LAST_NAME = 'Please enter valid last name';
+export const VALIDATION_CONTACT_NUMBER = 'Please enter valid contact number';
+export const VALIDATION_ADDRESS = 'Please enter valid address';
+export const VALIDATION_USERNAME = 'Please enter valid username';
+export const VALIDATION_PASSWORD = 'Please enter valid password';
+export const SUCCESS_SIGNUP = 'User registered successfully';
```
Comment (AccentureJace) [inline src/constants/auth/index.js:9]: separate this from VALIDATION by adding next line

File: src/constants/auth/index.js [hunk 5]
```diff
@@ -1 +1,13 @@
...
+export const SUCCESS_SIGNIN = 'Successfully logged in';
+export const INVALID_USERNAME = 'Invalid Username';
```
Comment (AccentureJace) [inline src/constants/auth/index.js:12]: separate this from SUCCESS by adding next line

File: src/constants/auth/index.js [hunk 6]
```diff
@@ -1 +1,13 @@
...
+export const INVALID_PASSWORD = 'Invalid password';
+export const ERROR_400 = 'User already logged in!';
+export const SERVER_ERROR = 'An error occurred. Please try again later.';
```
Comment (AccentureJace) [inline src/constants/auth/index.js]: ERROR_SERVER

File: src/pages/auth/Signup.jsx [hunk 7]
```diff
@@ -33,36 +42,28 @@ const Signup = () => {
...
                 last_name: values.last_name,
-                contact_number: values.contact_number,
                 address: values.address,
+                contact_number: values.contact_number,
                 username: values.username,
                 password: values.password,
             };
             const response = await authService.register(registerData);
-            if (response.status === 200) {
+            if (response.message === SUCCESS_SIGNUP) {
```
Comment (AccentureJace) [inline src/pages/auth/Signup.jsx]: utilize destructure

const { message, response } = response

File: src/pages/auth/Signup.jsx [hunk 8]
```diff
@@ -33,36 +42,28 @@ const Signup = () => {
...
                 form.resetFields();
                 setTimeout(() => {
-                    navigate({ PATH_LOGIN });
+                    navigate(PATH_LOGIN);
                 }, 1500);
-                toast.success('Register successful!');
+                toast.success(response.message);
             } else {
-                toast.error('An error occurred. Please try again later.');
+                toast.error(response.response.data.error.message);
```
Comment (AccentureJace) [inline src/pages/auth/Signup.jsx]: utilize destructure

const { message as ___ } = response.data.error

**Review by AccentureJace — CHANGES_REQUESTED** [review]
Please see request changes

**Review by AccentureJace — APPROVED** [review]
LGTM

**Review by AccentureAngelo — APPROVED** [review]

**Pull request merged**